kubernetes/kubernetes_service.go

```diff
@@ -236,3 +267,21 @@ func (in *IstioClient) getDeployments(namespace string, deploymentsChan chan dep
 	deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
 	deploymentsChan <- deploymentsResponse{deployments: deployments, err: err}
 }
+
+func (in *IstioClient) getDeployment(namespace string, deploymentName string) (*v1beta1.Deployment, error) {
+	return in.k8s.AppsV1beta1().Deployments(namespace).Get(deploymentName, emptyGetOptions)
+}
+
+func (in *IstioClient) getPods(namespace string, selector map[string]string, podsChan chan podsResponse) {
+	selectorQuery := labels.Set(selector).String()
+	pods, err := in.GetPods(namespace, selectorQuery)
+	podsChan <- podsResponse{pods: pods, err: err}
+}
+
+func (in *IstioClient) getServices(namespace string, selector map[string]string, serviceChan chan servicesResponse) {
+	cfg := config.Get()
+	serviceSelectorName := labels.Set(selector).Get(cfg.ServiceFilterLabelName)
+	serviceSelector := labels.Set{cfg.ServiceFilterLabelName: serviceSelectorName}
+	services, err := in.k8s.CoreV1().Services(namespace).List(meta_v1.ListOptions{LabelSelector: serviceSelector.String()})
+	serviceChan <- servicesResponse{services: services, err: err}
```

Comment:
Similar, here. The idea is that I get the service through the `app` label because there isn't a strong association between deployment and services.